### PR TITLE
#24903: supporting credibility interval and risk for variant results

### PR DESCRIPTION
**Pull request by victoralfaro-dotcms**
Adapting Bayesian API to include priors and to return credibility intervals as well as risk for each variant result.
Tests updated.

### CODE REVIEW

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/analytics/bayesian/BayesianAPI.java [hunk 1]
```diff
@@ -16,65 +19,36 @@
  */
 public interface BayesianAPI {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/analytics/bayesian/BayesianAPI.java:20]: DOc me

File: dotCMS/src/main/java/com/dotcms/analytics/bayesian/BayesianAPIImpl.java [hunk 2]
```diff
@@ -41,302 +43,459 @@
...
-        final Optional<BayesianResult> noopResult = detectNoop(input);
-        if (noopResult.isPresent()) {
-            return noopResult.get();
-        }
-
-        final List<VariantProbability> results = calcABCTesting(
-            input.control(),
-            input.variantPairs().get(0),
-            input.variantPairs().get(1));
+    @NotNull
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/analytics/bayesian/BayesianAPIImpl.java:64]: remove this

File: dotCMS/src/main/java/com/dotcms/analytics/bayesian/BayesianAPIImpl.java [hunk 3]
```diff
@@ -41,302 +43,459 @@
...
+     * @param input Bayesian input
+     * @return Bayesian result
      */
-    private double calcABTesting(final VariantInputPair control, final VariantInputPair test) {
-        return calcABTesting(
-            control.successes() + 1,
-            control.failures() + 1,
-            test.successes() + 1,
-            test.failures() + 1);
+    @NotNull
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/analytics/bayesian/BayesianAPIImpl.java:163]: renove this

File: dotCMS/src/main/java/com/dotcms/analytics/bayesian/BayesianAPIImpl.java [hunk 4]
```diff
@@ -41,302 +43,459 @@
...
     /**
-     * Given a {@link BetaDistribution} instance calculates density (pdf) elements.
+     * Resolves which variant could be considered as the winner of the test.
+     *
+     * @param value calculated probability that B (Test) beats A (Control)
+     * @param control control variant
+     * @param test test variant
+     * @return winner variant name
+     */
+    @NotNull
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/analytics/bayesian/BayesianAPIImpl.java:416]: remove this

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/analytics/bayesian/BayesianAPIImpl.java (hunk 4 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/analytics/bayesian/BayesianAPIImpl.java:475]: Removed.

File: dotCMS/src/main/java/com/dotcms/analytics/bayesian/BayesianAPIImpl.java (hunk 2 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/analytics/bayesian/BayesianAPIImpl.java:65]: Removed.

File: dotCMS/src/main/java/com/dotcms/analytics/bayesian/BayesianAPI.java (hunk 1 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/analytics/bayesian/BayesianAPI.java:20]: Dcoumented.

**Pull request merged**